scripts/zones/Altar_Room/mobs/Yagudo_Avatar.lua

```diff
@@ -0,0 +1,27 @@
+-----------------------------------
+-- Area: Altar Room 
+-- NPC: Stone Lid
+-- !pos -316.4390, 24.7654, 12.1590
+-----------------------------------
+local ID = require("scripts/zones/Altar_Room/IDs")
+mixins = {require("scripts/mixins/job_special")}
+require("scripts/globals/mobs")
+
+function onMobSpawn(mob)
+end
+
+function onMobDeath(mob, player, isKiller)
+    if  player:getQuestStatus(OTHER_AREAS_LOG,tpz.quest.id.otherAreas.A_MORAL_MANIFEST) == QUEST_ACCEPTED and player:getCharVar("moral") == 5 then
+        player:setCharVar("moral", 6)
+        player:delKeyItem(tpz.ki.VAULT_QUIPUS);
+    end
+    for i = ID.mob.YAGUDO_AVATAR + 1, ID.mob.YAGUDO_AVATAR + 8 do
+        DespawnMob(i)
+    end
+end
+
+function onMobDespawn(mob)
+    for i = ID.mob.YAGUDO_AVATAR + 1, ID.mob.YAGUDO_AVATAR + 8 do
+        DespawnMob(i)
+    end
+end
```

Comment:
You might want a check to see if these mobs are alive before despawning them to avoid errors.
something like:
```
if not GetMobByID(i):isDead() then
    DespawnMob(i)
end
```

someone else might have a better method to add, this is just a quick example.

Reply:
This actually shouldn't be a problem, and if it is we should make the core not let that be a problem. If I tell an entity to despawn that is already in the process of doing so it should just do nothing (as soon as they die they code sends them to the despawn state).